lib/util/git/index.ts

```diff
@@ -42,6 +43,50 @@ export { setPrivateKey } from './private-key';
 // TODO: fix upstream types https://github.com/steveukx/git-js/issues/704
 const ResetMode = (simpleGit.default as any).ResetMode as typeof _ResetMode;
 
+// Retry parameters
+const retryCount = 5;
+const delaySeconds = 3;
+const delayFactor = 2;
+
+// A generic wrapper for simpleGit.* calls to make them more fault-tolerant
+export async function gitRetry<T>(gitFunc: () => Promise<T>): Promise<T> {
+  let round = 0;
+  while (round <= retryCount) {
+    if (round > 0) {
+      logger.debug({ round }, 'gitRetry round');
+    }
+    try {
+      return await gitFunc();
+    } catch (err) {
+      logger.debug({ err }, `Git function thrown`);
+      try {
+        // Try to transform the Error to ExternalHostError
+        checkForPlatformFailure(err);
+        throw err;
+      } catch (errChecked) {
+        if (errChecked instanceof ExternalHostError) {
+          logger.debug(
+            { errChecked },
+            `ExternalHostError thrown in round ${
+              round + 1
+            } of ${retryCount} - retrying in the next round`
+          );
+        } else {
+          throw err;
+        }
+      }
+    }
+
+    const nextDelay = delayFactor ^ ((round - 1) * delaySeconds);
+    logger.debug({ nextDelay }, `Waiting ${nextDelay} seconds`);
```

Comment:
```suggestion
    logger.trace({ nextDelay }, `Delay next round`);
```
Not very useful message.